### PR TITLE
1.1.0 alpha perftest

### PR DESCRIPTION
**Pull request by khaledkbadr**

### CODE REVIEW

File: api/node/node_api_CreateContainer.go [hunk 1]
```diff
@@ -51,6 +52,10 @@ func (api NodeAPI) CreateContainer(w http.ResponseWriter, r *http.Request) {
 	var mounts = make([]mount, len(reqBody.Filesystems))
 	for idx, filesystem := range reqBody.Filesystems {
 		parts := strings.Split(filesystem, ":")
```
Comment (serboctor) [inline api/node/node_api_CreateContainer.go]: This should be removed to the CreateContainer type validator.

File: api/node/node_api_FileDownload.go [hunk 2]
```diff
@@ -20,14 +23,19 @@ func (api NodeAPI) FileDownload(w http.ResponseWriter, r *http.Request) {
...
 
-	w.Header().Set("Content-Type", "application/octet-stream")
-	w.WriteHeader(http.StatusOK)
 	if err := fs.Download(path, w); err != nil {
 		tools.WriteError(w, http.StatusInternalServerError, err)
+		return
 	}
+
+	w.Header().Set("Content-Type", "application/octet-stream")
+	w.WriteHeader(http.StatusOK)
```
Comment (grimpy) [inline api/node/node_api_FileDownload.go]: fs.Download already writes to http so should be done first no?

File: buildscripts/build-performance-test.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+TARGET=$PERFORMANCE_TEST/target
+
+rm -rf $PERFORMANCE_TEST
+mkdir -p $PERFORMANCE_TEST/src
+mkdir -p $TARGET/bin
+mkdir -p /tmp/archives
+
+
+cd $PERFORMANCE_TEST/src
+git clone --depth=1 https://github.com/maxux/lddcopy.git
```
Comment (grimpy) [inline buildscripts/build-performance-test.sh]: use the tools.sh and use ensure

File: buildscripts/build-performance-test.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+cd $PERFORMANCE_TEST/src/fio
+./configure && make
+cp fio $TARGET/bin
+
+cd $PERFORMANCE_TEST/src/nbd-3.15.2
+./configure && make
+cp nbd-client $TARGET/bin
+
+bash $PERFORMANCE_TEST/src/lddcopy/lddcopy.sh $TARGET/bin/fio $TARGET
```
Comment (grimpy) [inline buildscripts/build-performance-test.sh]: just use lddcopy after using the ensure

File: tests/bd-performance.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        # Create and setup the test container
+        fioFlist = "https://hub.gig.tech/gig-official-apps/performance-test.flist"
+        createContainer(resourcepoolserver, api, nodeID, [fss], fioFlist, "bptest")
+
+        # Load nbd kernel module
+        nodeClient = g8core.Client(nodeIP)
+        nodeClient.bash("modprobe nbd").get()
+
+        filenames = nbdClientConnect(api, nodeID, "bptest", nbdConfig)
+        FIOCOMMAND = {
```
Comment (grimpy) [inline tests/bd-performance.py]: shoudl not be capitlized

**Pull request merged**